app/waf/test/acceptance/adc.controller.acceptance.ts

```diff
@@ -423,9 +446,281 @@ describe('AdcController', () => {
+    },
+  );
+
+  it(
+    'post ' + prefix + '/adcs: create ADC HW with wrong AS3 extension response',
+    async function() {
+      await givenAdcData(wafapp, {
+        trustedDeviceId: 'abcdefg',
+      });
+
+      const adc = createAdcObject({
+        type: 'HW',
+        management: {
+          ipAddress: '1.2.3.4',
+          tcpPort: 100,
+          username: 'admin',
+          password: 'admin',
+          rootPass: 'default',
+        },
+      });
+
+      let id = uuid();
+      trustStub.returns({
+        devices: [
+          {
+            targetUUID: id,
+            targetHost: '1.2.3.4',
+            state: 'CREATED',
+          },
+        ],
+      });
+
+      queryStub.onCall(0).returns({
+        devices: [
+          {
+            targetUUID: id,
+            targetHost: '1.2.3.4',
+            state: 'PENDING',
+          },
+        ],
+      });
+
+      queryStub.returns({
+        devices: [
+          {
+            targetUUID: id,
+            targetHost: '1.2.3.4',
+            state: 'ACTIVE',
+          },
+        ],
+      });
+
+      queryExtensionsStub.returns([]);
+
+      let response = await client
+        .post(prefix + '/adcs')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send(adc)
+        .expect(200);
+
+      expect(response.body.adc).to.containDeep(toJSON(adc));
+
+      await sleep(100);
+
+      response = await client
+        .get(prefix + '/adcs/' + response.body.adc.id)
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .expect(200);
+
+      expect(response.body.adc.status).to.equal('INSTALLERROR');
+    },
+  );
+
+  it(
+    'post ' + prefix + '/adcs: create ADC HW with query extension exception',
+    async function() {
+      await givenAdcData(wafapp, {
+        trustedDeviceId: 'abcdefg',
+      });
+
+      const adc = createAdcObject({
+        type: 'HW',
+        management: {
+          ipAddress: '1.2.3.4',
+          tcpPort: 100,
+          username: 'admin',
+          password: 'admin',
+          rootPass: 'default',
+        },
+      });
+
+      let id = uuid();
+      trustStub.returns({
+        devices: [
+          {
+            targetUUID: id,
+            targetHost: '1.2.3.4',
+            state: 'CREATED',
+          },
+        ],
+      });
+
+      queryStub.returns({
+        devices: [
+          {
+            targetUUID: id,
+            targetHost: '1.2.3.4',
+            state: 'ACTIVE',
+          },
+        ],
+      });
+
+      queryExtensionsStub.throws(new Error('query-not-working'));
+
+      let response = await client
+        .post(prefix + '/adcs')
+        .set('X-Auth-Token', ExpectedData.userToken)
+        .set('tenant-id', ExpectedData.tenantId)
+        .send(adc)
+        .expect(200);
+
+      expect(response.body.adc).to.containDeep(toJSON(adc));
+
+      await sleep(50);
```

Comment:
Left it here, I'll change it away.

Reply:
After checkAndWait() can support error path, we will not need to sleep here.